### PR TITLE
Allow overriding association's cascade

### PR DESCRIPTION
**Pull request by malarzm**
We were badly bitten by `cascade` set on an association we did not expect it to have. The association is dynamically mapped using a `loadClassMetadata` event by a vendor with no way of hooking into it. As we're really careful with persisting stuff I'd like to have an option to remove vendor's cascade and instead take care of it within our codebase.

Additionally I've listed what can actually be overridden as I was surprised it's that limited :)

### CODE REVIEW

File: tests/Tests/ORM/Mapping/ClassMetadataTest.php [hunk 1]
```diff
@@ -870,13 +870,23 @@ public function testAssociationOverrideKeepsDeclaringClass(): void
     {
         $cm = new ClassMetadata(Directory::class);
         $cm->mapManyToOne(['fieldName' => 'parentDirectory', 'targetEntity' => Directory::class, 'cascade' => ['remove'], 'declared' => Directory::class]);
-        $cm->setAssociationOverride('parentDirectory', ['cascade' => '']);
+        $cm->setAssociationOverride('parentDirectory', ['cascade' => ['remove']]);
```
Comment (SenseException) [inline tests/Tests/ORM/Mapping/ClassMetadataTest.php:873]: Why this change in the test?

**Review by SenseException — COMMENTED** [review]
Kind of weird that everything can be overridden but not `cascade`. I assume this was forgotten to be implemented, but I'm not sure if this had a reason.

File: tests/Tests/ORM/Mapping/ClassMetadataTest.php (hunk 1 above)
Comment (malarzm) [reply tests/Tests/ORM/Mapping/ClassMetadataTest.php:873]: Previously `cascade` was ignored, now it's leading to an error as it's supposed to be an array of strings. I reckon this test was originally done to pass something and see if nothing (declaring class) breaks

**Comment by malarzm** [comment]
> I assume this was forgotten to be implemented, but I'm not sure if this had a reason.

I gave it some thought and couldn't think of any reason this would be omitted on purpose. My guess is that nobody needed it as overriding association is not very common: people either configure things themselves or roll with vendors. I don't like vendors shooting me in the foot :D

**Review by SenseException — APPROVED** [review]

**Comment by malarzm** [comment]
@SenseException can we treat this as bugfix and still merge to 3.x?

**Comment by SenseException** [comment]
I guess being the only option that can't be overridden could be interpreted as a bug. I guess this is a situation that can be argued about. :grin: 

I would give an okay to handle this as a bugfix.

**Review by greg0ire — APPROVED** [review]
You didn't have to change any tests, so yeah, it should be fine 👍
